odl/phantom/geometric_phantoms.py

```diff
@@ -0,0 +1,189 @@
+﻿# Copyright 2014-2016 The ODL development group
+#
+# This file is part of ODL.
+#
+# ODL is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# ODL is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with ODL.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Phantoms given by simple geometric objects such as cubes or spheres."""
+
+# Imports for common Python 2/3 codebase
+from __future__ import print_function, division, absolute_import
+from future import standard_library
+standard_library.install_aliases()
+
+import numpy as np
+
+__all__ = ('cuboid', 'indicate_proj_axis')
+
+
+def cuboid(discr_space, begin=0.25, end=0.75):
+    """Rectangular cuboid.
+
+    Parameters
+    ----------
+    discr_space : `DiscretizedSpace`
+        Discretized space in which the phantom is supposed to be created.
+    begin : array-like or `float` in [0, 1]
+        The lower left corner of the cuboid within the space grid relative
+        to the extend of the grid.
+    end : array-like or `float` in [0, 1]
+        The upper right corner of the cuboid within the space grid relative
+        to the extend of the grid.
+
+    Returns
+    -------
+    phantom : `LinearSpaceVector`
+        Returns an element in ``discr_space``
+
+    Examples
+    --------
+    >>> import odl
+    >>> space = odl.uniform_discr(0, 1, 6, dtype='float32')
+    >>> print(cuboid(space, 0.5, 1))
+    [0.0, 0.0, 0.0, 1.0, 1.0, 1.0]
+    >>> space = odl.uniform_discr([0, 0], [1, 1], [4, 6], dtype='float32')
+    >>> print(cuboid(space, [0.25, 0], [0.75, 0.5]))
+    [[0.0, 0.0, 0.0, 0.0, 0.0, 0.0],
+     [1.0, 1.0, 1.0, 0.0, 0.0, 0.0],
+     [1.0, 1.0, 1.0, 0.0, 0.0, 0.0],
+     [0.0, 0.0, 0.0, 0.0, 0.0, 0.0]]
+    """
+    ndim = discr_space.ndim
+    shape = discr_space.shape
+
+    if np.isscalar(begin):
+        begin = (begin,) * ndim
+    if np.isscalar(end):
+        end = (end,) * ndim
```

Comment:
This is in contrast to all the other places where similar input parameters are handled. We should expect sequences of correct size in nD and 1-sequence or float in 1D.